include/aws/common/linked_list.h

```diff
@@ -98,10 +176,15 @@ AWS_STATIC_IMPL struct aws_linked_list_node *aws_linked_list_prev(const struct a
 AWS_STATIC_IMPL void aws_linked_list_insert_after(
     struct aws_linked_list_node *after,
     struct aws_linked_list_node *to_add) {
+    AWS_PRECONDITION(aws_linked_list_node_next_is_valid(after));
+    AWS_PRECONDITION(to_add != NULL);
     to_add->prev = after;
     to_add->next = after->next;
     after->next->prev = to_add;
     after->next = to_add;
+    AWS_POSTCONDITION(aws_linked_list_node_next_is_valid(after));
+    AWS_POSTCONDITION(aws_linked_list_node_prev_is_valid(to_add));
+    AWS_POSTCONDITION(aws_linked_list_node_next_is_valid(to_add));
```

Comment:
As long as you're going to have `is_connected`, you may as well check it as a pre and post condition on all of the functions that modify the list, right?

Reply:
This specific function acts locally on nodes and is agnostic of the list that those nodes are in, so I think it would be unnecessary to assert anything about the list. In contrast, in `aws_linked_list_push_back` and `aws_linked_list_push_front` connectivity is asserted.